WN10.P4 Support IOpenApiDocumentProvider in DI container (#35428)
dotnet/AspNetCore.Docs: aspnetcore/release-notes/aspnetcore-10/includes/doc-provider-in-di.md

## Support for IOpenApiDocumentProvider in the DI container.

> ASP.NET Core in .NET 10 supports `IOpenApiDocumentProvider` in the dependency injection (DI) container. Developers can inject it into their applications and use it to access the OpenAPI document. This approach is useful for accessing OpenAPI documents outside the context of HTTP requests, such as in background services or custom middleware.

Reviewer: ```suggestion
ASP.NET Core in .NET 10 supports [IOpenApiDocumentProvider](https://source.dot.net/#Microsoft.AspNetCore.OpenApi/Services/IOpenApiDocumentProvider.cs) in the dependency injection (DI) container. Developers can inject `IOpenApiDocumentProvider` into their apps and use it to access the OpenAPI document. This approach is useful for accessing OpenAPI documents outside the context of HTTP requests, such as in background services or custom middleware.
```

> Previously, running application startup logic without launching an HTTP server could be done by using `HostFactoryResolver` with a no-op `IServer` implementation. The new feature simplifies this process by providing a streamlined API inspired by Aspire's <xref:Aspire.Hosting.Publishing.IDistributedApplicationPublisher>.

Reviewer: I agree with CoPilot on that

For more information, see [dotnet/aspnetcore #61463](https://github.com/dotnet/aspnetcore/pull/61463)

> <!--[!INCLUDE[](~/release-notes/aspnetcore-10/includes/doc-provider-in-di.md)] -->

Reviewer: Move that to WN so we can see the build. New policy..

## Support for IOpenApiDocumentProvider in the DI container.

ASP.NET Core in .NET 10 supports [IOpenApiDocumentProvider](https://source.dot.net/#Microsoft.AspNetCore.OpenApi/Services/IOpenApiDocumentProvider.cs) in the dependency injection (DI) container. Developers can inject `IOpenApiDocumentProvider` into their apps and use it to access the OpenAPI document. This approach is useful for accessing OpenAPI documents outside the context of HTTP requests, such as in background services or custom middleware.

> Previously, running application startup logic without launching an HTTP server could be done by using `HostFactoryResolver` with a no-op `IServer` implementation. The new feature simplifies this process by providing a streamlined API inspired by Aspire's <xref:Aspire.Hosting.Publishing.IDistributedApplicationPublisher>.

Reviewer 2: I think this paragraph is describing functionality that was not actually implemented because of concerns raised in the API review. The `IOpenApiDocumentProvider` interface does not change the way OpenAPI documents are generated at build time. So I recommend simply removing this paragraph.  @captainsafia can you pls confirm?
```suggestion
```

> Previously, running application startup logic without launching an HTTP server could be done by using `HostFactoryResolver` with a no-op `IServer` implementation. The new feature simplifies this process by providing a streamlined API inspired by Aspire's <xref:Aspire.Hosting.Publishing.IDistributedApplicationPublisher>For more information, see [Aspire Documentation](https://aspire.example.com/docs/IDistributedApplicationPublisher).

Author: ```suggestion
Previously, running application startup logic without launching an HTTP server could be done by using `HostFactoryResolver` with a no-op `IServer` implementation. The new feature simplifies this process by providing a streamlined API inspired by Aspire's <xref:Aspire.Hosting.Publishing.IDistributedApplicationPublisher>. For more information, see [Aspire Documentation](https://aspire.example.com/docs/IDistributedApplicationPublisher).
```

> Previously, running application startup logic without launching an HTTP server could be done by using `HostFactoryResolver` with a no-op `IServer` implementation. The new feature simplifies this process by providing a streamlined API inspired by Aspire's <xref:Aspire.Hosting.Publishing.IDistributedApplicationPublisher>. For more information, see [Aspire Documentation](https://aspire.example.com/docs/IDistributedApplicationPublisher).
> 
> For more information, see [dotnet/aspnetcore #61463](https://github.com/dotnet/aspnetcore/pull/61463)

Author: ```suggestion
Previously, running application startup logic without launching an HTTP server could be done by using `HostFactoryResolver` with a no-op `IServer` implementation. The new feature simplifies this process by providing a streamlined API inspired by Aspire's <xref:Aspire.Hosting.Publishing.IDistributedApplicationPublisher>.

For more information, see [Aspire Documentation](https://aspire.example.com/docs/IDistributedApplicationPublisher) and [dotnet/aspnetcore #61463](https://github.com/dotnet/aspnetcore/pull/61463)
```
